src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -445,6 +653,6 @@ def __aiter__(self) -> AsyncIterable[ResponseType]:
             # If the read operation failed, Core should explain why.
             await self._raise_for_status()
             # If no exception raised, there is something wrong internally.
-            assert False, 'Read operation failed with StatusCode.OK'
+            raise RuntimeError('Read operation failed with StatusCode.OK')
```

Comment:
I'm a bit lost on how the end of the reading - so message exhaustion - is signaled to the consumer. Seems that a `None` value returned by the`_read` is a perfectly valid value that is later on used by the `_fetch_stream_responses` for knowing when the iterator needs to stop.

But this logic seems completely broken - most likely I'm missing something - when the primitive `read` is used, any `None` value would be considered as abnormal termination of the RPC.

What I'm missing?

Reply:
Changed to use a designated `EOF` value.

For most encoder/decoder, the result should not be `None`. It works if we agree with the assumption, and treat `None` as a termination signal.

It seems supporting `None` message is what you are asking here. As discussed before, we could use a special `EOF` value to notify ending of messages.

Reply:
For me `None` should be also a valid terminal value, but also I see the value of EOF which is more explicit. I'm fine with either of them.

My main worry was more about the exception raised yes or yes when a None value was returned when an explicit `read()` was done,  which you have addressed.